Jenkinsfile

```diff
@@ -0,0 +1,173 @@
+#!groovy
+/*
+ * This Jenkinsfile is intended to run on https://ci.jenkins-ci.org and may fail anywhere else.
+ * It makes assumptions about plugins being installed, labels mapping to nodes that can build what is needed, etc.
+ */
+
+// TEST FLAG - to make it easier to turn on/off unit tests for speeding up access to later stuff.
+def runTests = true
+
+// Only keep the 10 most recent builds.
+properties([[$class: 'BuildDiscarderProperty', strategy: [$class: 'LogRotator',
+                                                          numToKeepStr: '10']]])
+
+// TODO: Once https://github.com/jenkinsci/packaging/pull/34 is merged, switch to master branch.
+String packagingBranch = (binding.hasVariable('packagingBranch')) ? packagingBranch : 'oss-dockerized-tests'
+
+node('java') {
+
+    // Add timestamps to logging output.
+    wrap([$class: 'TimestamperBuildWrapper']) {
+
+        // First stage is actually checking out the source. Since we're using Multibranch
+        // currently, we can use "checkout scm".
+        stage "Checkout source"
+
+        checkout scm
+
+        // Now run the actual build.
+        stage "Build and test"
+
+        // We're wrapping this in a timeout - if it takes more than 180 minutes, kill it.
+        timeout(time: 180, unit: 'MINUTES') {
+            // See below for what this method does - we're passing an arbitrary environment
+            // variable to it so that JAVA_OPTS and MAVEN_OPTS are set correctly.
+            withMavenEnv(["JAVA_OPTS=-Xmx1536m -Xms512m -XX:MaxPermSize=1024m",
+                          "MAVEN_OPTS=-Xmx1536m -Xms512m -XX:MaxPermSize=1024m"]) {
+                // Actually run Maven!
+                // The -Dmaven.repo.local=${pwd()}/.repository means that Maven will create a
+                // .repository directory at the root of the build (which it gets from the
+                // pwd() Workflow call) and use that for the local Maven repository.
+                sh "mvn -Pdebug -U clean install ${runTests ? '-Dmaven.test.failure.ignore=true -Dconcurrency=1' : '-DskipTests'} -V -B -Dmaven.repo.local=${pwd()}/.repository"
+            }
+        }
+
+        // Once we've built, archive the artifacts and the test results.
+        stage "Archive artifacts and test results"
+
+        step([$class: 'ArtifactArchiver', artifacts: '**/target/*.jar, **/target/*.war, **/target/*.hpi', excludes: null, fingerprint: true])
+        if (runTests) {
+            step([$class: 'JUnitResultArchiver', healthScaleFactor: 20.0, testResults: '**/target/surefire-reports/*.xml'])
+        }
+    }
+}
+
+def debFileName
+def rpmFileName
+def suseFileName
+
+// Run the packaging build on a node with the "docker" label.
+node('docker') {
+    // Add timestamps to logging output.
+    wrap([$class: 'TimestamperBuildWrapper']) {
```

Comment:
Since you are always using `TimestamperBuildWrapper` this idiom could be interesting:

```
timestampedNode('docker') {
    // real stuff here
}

def timestampedNode(String label, Closure body) {
    node(label) {
        wrap([$class: 'TimestamperBuildWrapper']) {
            body();
        }
    }
}
```